controllers/volsync/vshandler.go

```diff
@@ -744,17 +761,24 @@ func (v *VSHandler) deleteLocalRDAndRS(rd *volsyncv1alpha1.ReplicationDestinatio
 	lrs := &volsyncv1alpha1.ReplicationSource{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      getLocalReplicationName(rd.GetName()),
-			Namespace: v.owner.GetNamespace(),
+			Namespace: rd.GetNamespace(),
 		},
 	}
 
-	err = v.client.Get(v.ctx, types.NamespacedName{
+	lrsNamespacedName := types.NamespacedName{
 		Name:      lrs.GetName(),
 		Namespace: lrs.GetNamespace(),
-	}, lrs)
+	}
+
+	err = v.client.Get(v.ctx, lrsNamespacedName, lrs)
 	if err != nil {
 		if kerrors.IsNotFound(err) {
-			return v.deleteLocalRD(getLocalReplicationName(rd.GetName()))
+			lrdNamespacedName := types.NamespacedName{
+				Namespace: rd.GetNamespace(),
+				Name:      getLocalReplicationName(lrs.GetName()),
```

Comment:
This is a bug: double "-local".  Perhaps this contributed to the volsync regression test failures?

Reply:
Yes, that's a bug, but it didn't contribute to the regression.